src/requests/adapters.py

```diff
@@ -389,7 +390,7 @@ def request_url(self, request, proxies):
             proxy_scheme = urlparse(proxy).scheme.lower()
             using_socks_proxy = proxy_scheme.startswith("socks")
 
-        url = request.path_url
+        url = re.sub("^/+", "/", request.path_url)
```

Comment:
As mentioned on https://github.com/urllib3/urllib3/issues/3352 this could also be

```python
url = f"/{request.path_url.lstrip('/')}"
```

I could benchmark these but I don't particularly care what the implementation is. I just threw this together to show that it can be fixed

Reply:
It looks like the f-string (Python 3.9-3.12 tested) is ~4x faster but we're talking on the scale of nanoseconds so it's basically moot. I'd vote the f-string for readability, but don't have a strong opinion.

Reply:
Yeah, I'm also happy to shove this into a branch too like 

```python
if path.startswith('//'):
```

To make it clearer that we only care about the separator being repeated. What I want is clarity in the reader as to why we're doing this. My old school brain things the regexp is clearer and the f-string looks sus but  that's just my opinion and I'm not holding it closely

Reply:
Yeah, branching seems fine to me too.

Reply:
Did both (f string and branch)